auslib/web/public/client.py

```diff
@@ -85,7 +85,8 @@ def getQueryFromURL(url):
     query['osVersion'] = urllib.unquote(query['osVersion'])
     ua = request.headers.get('User-Agent')
     query['headerArchitecture'] = getHeaderArchitecture(query['buildTarget'], ua)
-    query['force'] = (int(query.get('force', 0)) == 1)
+    force = query.get('force')
+    query['force'] = {'1': SUCCEED, '0': FAIL}.get(force)
```

Comment:
How there are two specific options to always get a fallback release or always get the up-to-date release, maybe the force values  can be different of zero (i.e 1 succed, 2 fail), and ignore all other "false" cases. Maybe change query string name to something more close to this context.
What you think?

Reply:
I don't have strong opinions on the API itself. However, I assumed we had to maintain backwards compatibility with Firefoxes that are already "in the wild", which will continue passing `force=1`. But my inclination is that "force" isn't the worst name, since we're forcing the random number to either be successful (not be throttled) or fail (be throttled). And of course we have to distinguish "force to succeed" and "force to fail" from "no forcing". It seems more natural to me to have 0 represent force to fail and no query argument, or an empty string, represent "no forcing", rather than 2 represent force to fail and 0 represent "no forcing".

Of course, internally we have to have three distinct values to represent these three requests, and for those I chose to define two arbitrary constants `SUCCESS` and `FAIL`, neither of which are falsy, and use `None` for "no forcing". I could have just as easily used `True` and `False`, but I think it's good to define a unique enumeration for a unique use like this one.

Reply:
Oh!Yeah, makes all sense the querystring name not change, sorry!
I commented mainly by the "force=0", this seems say "do not force anything".

Reply:
I'd much prefer `force=succeed` and `force=fail`, but I'm not sure how to make that happen. I think `force=1` and `force=fail` would be a bit strange but I'm willing to go for it if you are. `force=1` and `force=-1` might be a compromise.

Reply:
I like force=-1